perceval/utils/conversion.py

```diff
@@ -59,13 +59,18 @@ def samples_to_probs(sample_list: BSSamples) -> BSDistribution:
 
 def probs_to_sample_count(probs: BSDistribution, count: int = None, **kwargs) -> BSCount:
     count = _deduce_count(count, **kwargs)
+    if count < 1:
+        return BSCount()
     perturbed_dist = {state: max(prob + np.random.normal(scale=(prob * (1 - prob) / count) ** .5), 0)
                       for state, prob in probs.items()}
-    prob_sum = sum(prob for prob in perturbed_dist.values())
+    prob_sum = sum(perturbed_dist.values())
     if prob_sum == 0:
         return samples_to_sample_count(probs_to_samples(probs, count))
     fac = 1 / prob_sum
     perturbed_dist = {key: fac * prob for key, prob in perturbed_dist.items()}  # Renormalisation
+    if max(perturbed_dist.values()) * count < 1:
+        return samples_to_sample_count(probs_to_samples(probs, count))
+
     results = BSCount()
     for state in perturbed_dist:
         results.add(state, round(perturbed_dist[state] * count))
```

Comment:
Oh ok that's why...

Reply:
I still do not get it

Reply:
This function converts an exact probability distribution to a sample count, _as if it was really sampled_.
The perturbation in the input exact distribution simulates this sampling.

Reply:
@ericbrts, just one question - I fail to see how the condition (L71) `max(perturbed_dist.values()) * count < 1` removes the exception and quantifies low sample count.

Reply:
This one I can answer : for every key s.t. ```perturbated_dist.value * count < 1``` you will get a 0 simulated sample count.

Thus if max(XXX)*count < 1, there will be no entry at all.

Reply:
Anyway, after one night thinking of it, I'd rather replace ```results[random.choice(list(results.keys()))] += diff``` with ```results[random.choice(list(probs.keys()))] += diff```.

probs is not empty (checked by ```if prob_sum == 0```) and this would avoid real sampling when we said we would simulate it.

But this doesn't really change the beahavior, so this remark is "just a thought".

Reply:
(and I hope that at some time someone will say "Don't worry Benoit, the random seed is freezed during tests", because debugging a non-reproductable unit test looks like hazing)

Reply:
The seed isn't fixed during auto tests - and we've got (rare) issues because of that.